AuditLog Resource addition

CoreyShupe opened the pull request:
Addresses https://github.com/princesslana/ErisCasper.java/issues/123

CoreyShupe committed:
Route update.

CoreyShupe committed:
Merge branch 'webhook-resource'

# Conflicts:
#	src/main/java/com/github/princesslana/eriscasper/rest/ChannelRoute.java
#	src/main/java/com/github/princesslana/eriscasper/rest/Route.java

CoreyShupe committed:
Audit log resource implementation.
Addresses https://github.com/princesslana/ErisCasper.java/issues/123

CoreyShupe committed:
Spotless apply

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/rest/AuditLogRoute.java:
```diff
@@ -0,0 +1,46 @@
+  }
+
+  /**
+   * Warning, this can and may throw an error. If it does please paste your console log into an
```
Well, this is true of anything isn't it?

CoreyShupe replied:
Especially true of this. Discord hasn't documented this well and I've run into undocumented incidents for this specific resource. It's more error prone so I felt it necessary to leave a comment about it.

ianagbip1oti replied:
Sorry, not a fan of this comment. More so the thing about filing an issue report (in areas where this comment isn't, I shouldn't file an issue report?).

An improvement would be something like 'These routes are not documented well in the Discord API documentation'

CoreyShupe replied:
Alright, I'll change the commenting, didn't really know how to word it as to avoid what you pointed out in the first statement. I'll reword it.

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/rest/auditlog/GetGuildAuditLog.java:
```diff
@@ -0,0 +1,37 @@
+ */
+@Value.Immutable
+public interface GetGuildAuditLog {
+  /*
```
This just duplicates what is at the linked url in the javadoc?

CoreyShupe replied:
yeap, noticed it late, removed

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/util/AuditLogEventType.java:
```diff
@@ -0,0 +1,64 @@
+
+  public static AuditLogEventType of(int code) {
+    AuditLogEventType type;
+    if (code > 72 || code < 1 || (type = types[code - 1]) == null) {
```
Don't like the assignment in the if statement.

Why do we use 72 instead of types.length?

CoreyShupe replied:
Yeah I need to fix the if statement, and I'll change it, forgot about that piece.

CoreyShupe replied:
The if statement is there to keep it in one if statement rather than 2.

CoreyShupe committed:
Channel route naming fix.

CoreyShupe committed:
Removed unnecessary comments and removed path from method.

CoreyShupe committed:
Format update, along with comment rewording.
Changes `of` in AuditLogEventType to split into 2 exceptions and remove the inner if statement.
Rewords the comment in AuditLogRoute.

CoreyShupe committed:
</grammar_nit> (Lana)

CoreyShupe committed:
Spotless

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/rest/auditlog/GetGuildAuditLog.java:
```diff
@@ -0,0 +1,31 @@
+ *     https://discordapp.com/developers/docs/resources/audit-log#get-guild-audit-log-query-string-parameters</a>
+ */
+@Value.Immutable
+public interface GetGuildAuditLog {
```
Haven't we been adding a Request suffix to these?

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/util/AuditLogEventType.java:
```diff
@@ -0,0 +1,68 @@
+      throw new IllegalArgumentException(
+          "Illegal code `" + code + "` found. Please refer to a code from 1-" + types.length + ".");
+    }
+    AuditLogEventType type;
```
This declaration doesn't have to be on it's own line

CoreyShupe replied:
I don't have an explanation for this, idk why I did this. I'm probably just really tired right now.

CoreyShupe committed:
Naming consistency.

CoreyShupe committed:
Fixed type declaration being on a different line.

CoreyShupe committed:
Class reference issue.

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/rest/AuditLogRoute.java:
```diff
@@ -0,0 +1,37 @@
+import com.github.princesslana.eriscasper.rest.auditlog.GetGuildAuditLogRequest;
+
+public class AuditLogRoute {
+  // https://discordapp.com/developers/docs/resources/audit-log#get-guild-audit-log
```
Should this be here?

CoreyShupe replied:
Doesn't have to be, I'll remove it. I think I added it in for myself since I was coming back to it just due to the parsing complexity.

CoreyShupe committed:
Simple comment removal.
